core/src/main/scala/gwi/s8/DagFSM.scala

```diff
@@ -115,11 +115,18 @@ class DagFSM(
       stay()
   }
 
+  onTermination {
+    case _ => cancellables.foreach(_.cancel())
+  }
+
   def applyEvent(event: DagStateEvent, dagState: DagFSMState): DagFSMState = event match {
     case e =>
       dagState.updated(e)
   }
 
+  override def onRecoveryCompleted() {
+    cancellables = schedulePartitionCheck(stateData) ++ cancellables
+  }
```

Comment:
@pribor I don't understand this, shouldn't `schedulePartitionCheck` be called also on initialization, not just on recavery?

Reply:
In the case of PersistentFSM it is called always even if there is no previous state/events, check tests..

Reply:
https://doc.akka.io/docs/akka/2.5.4/scala/persistence.html#recovery-status

Reply:
Nice one, this would save me a lot of troubles,I was struggling with this problem a lot. Nice work !